src/components/smartSearch/filters/Random/index.tsx

```diff
@@ -0,0 +1,144 @@
+import { FormattedMessage as Msg } from 'react-intl';
+import { Box, Button, Divider, MenuItem, Typography } from '@material-ui/core';
+import { FormEvent, useEffect, useState } from 'react';
+
+import { getQuantityWithConfig } from 'components/smartSearch/utils';
+import StyledNumberInput from '../../inputs/StyledNumberInput';
+import StyledSelect from '../../inputs/StyledSelect';
+import useSmartSearchFilter from 'hooks/useSmartSearchFilter';
+import { NewSmartSearchFilter , OPERATION, QUANTITY, RandomFilterConfig, SmartSearchFilterWithId, ZetkinSmartSearchFilter } from 'types/smartSearch';
+
+const shouldGenerateSeed = (
+    filter: ZetkinSmartSearchFilter<RandomFilterConfig>,
+    initialFilter: SmartSearchFilterWithId<RandomFilterConfig>) => {
+    const isInt = (num: number) => num % 1 === 0;
+    const currentSize = filter.config.size;
+    const initialSize = initialFilter.config.size;
+    return (
+        (isInt(currentSize) !== isInt(initialSize)) ||
+        (isInt(currentSize) && isInt(initialSize) && currentSize > initialSize) ||
+        (!isInt(currentSize) && !isInt(initialSize) && currentSize > initialSize)
+    );
+};
+
+interface RandomProps {
+    filter:  SmartSearchFilterWithId<RandomFilterConfig> |  NewSmartSearchFilter ;
+    onSubmit: (filter: SmartSearchFilterWithId<RandomFilterConfig> | ZetkinSmartSearchFilter<RandomFilterConfig>) => void;
+    onCancel: () => void;
+}
+
+const Random = ({ onSubmit, onCancel, filter: initialFilter }: RandomProps): JSX.Element => {
+    const { filter, setConfig, setOp } = useSmartSearchFilter<RandomFilterConfig>(initialFilter, { seed: Math.random().toString(), size : 0.5 });
+    const quantity = getQuantityWithConfig(filter.config.size);
```

Comment:
I like this, and lets keep it, but I think it might be a premature abstraction. I don't think there are any other filters that support percentages in Zetkin today.

Reply:
Yes! I didn't abstract the other parts (yaml, new component) because of AHA, but decided to do this to separate clearly the display value which also goes to translation, vs the value that was going to the API. More for clarity in my head than anything else.